components/renku_data_services/storage/db.py

```diff
@@ -137,9 +138,57 @@ async def delete_storage(self, storage_id: str, user: base_models.APIUser) -> No
 
             if storage is None:
                 return
-            if not await self.gitlab_client.filter_projects_by_access_level(
-                user, [storage[0].project_id], base_models.GitlabAccessLevel.ADMIN
-            ):
+            if not await self.filter_projects_by_access_level(user, [storage[0].project_id], authz_models.Role.OWNER):
                 raise errors.Unauthorized(message="User does not have access to this project")
 
             await session.delete(storage[0])
+
+
+class StorageRepository(BaseStorageRepository):
+    """Repository for V1 cloud storage."""
+
+    def __init__(
+        self,
+        gitlab_client: base_models.GitlabAPIProtocol,
+        session_maker: Callable[..., AsyncSession],
+    ):
+        super().__init__(session_maker)
+        self.gitlab_client = gitlab_client
+
+    async def filter_projects_by_access_level(
+        self, user: base_models.APIUser, project_ids: list[str], minimum_access_level: authz_models.Role
+    ) -> list[str]:
+        """Get a list of projects of which the user is a member with a specific access level."""
+        gitlab_access_level = (
+            base_models.GitlabAccessLevel.ADMIN
+            if minimum_access_level == authz_models.Role.OWNER
+            else base_models.GitlabAccessLevel.MEMBER
+        )
+
+        return await self.gitlab_client.filter_projects_by_access_level(user, project_ids, gitlab_access_level)
+
+
+class StorageV2Repository(BaseStorageRepository):
+    """Repository for V2 cloud storage."""
+
+    def __init__(
+        self,
+        project_authz: IProjectAuthorizer,
+        session_maker: Callable[..., AsyncSession],
+    ):
+        super().__init__(session_maker)
+        self.project_authz: IProjectAuthorizer = project_authz
+
+    async def filter_projects_by_access_level(
```

Comment:
I think in v2 we can be smarter about this and don't have to get all projects a user has access to and match against the storage, and just get auth directly for the storages?
Though that probably depends on the authz work @olevski is doing, so maybe we need to improve this in a follow-up issue?
The main reason to get all projects instead of checking individually in v1 is that the gitlab api is slow and I'd rather just send 1 request than lots of requests, but for v2 we shouldn't have that limitation